go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -2158,3 +2166,106 @@ func generateIPFragmentReassemblyFlow(ofPortPhys string) []string {
+func deleteMasqueradeResources(link netlink.Link, staleMasqueradeIPs *config.MasqueradeIPsConfig) error {
+	var subnets []*net.IPNet
+	var neighborIPs []net.IP
+	if config.IPv4Mode && staleMasqueradeIPs.V4HostMasqueradeIP != nil {
+		// Delete any stale masquerade IP from external bridge.
+		hostMasqIPNet, err := util.LinkAddrGetIPNet(link, staleMasqueradeIPs.V4HostMasqueradeIP)
+		if err != nil {
+			return fmt.Errorf("unable to get IPNet from link %s: %v", link, err)
+		}
+		if hostMasqIPNet != nil {
+			if err := util.LinkAddrDel(link, hostMasqIPNet); err != nil {
+				return fmt.Errorf("failed to remove masquerade IP from bridge %s: %w", link, err)
+			}
+		} else {
+			// This is to make sure that we don't delete anything if stale HostMasqueradeIP does not
+			// exist on gateway bridge. This will further stop deletion of resources if default stale
+			// masquerade subnet is being used for some other reason.
+			return nil
+		}
+		_, masqIPNet, err := net.ParseCIDR(fmt.Sprintf("%s/32", staleMasqueradeIPs.V4OVNMasqueradeIP.String()))
+		if err != nil {
+			return fmt.Errorf("failed to parse V4OVNMasqueradeIP %s: %v", staleMasqueradeIPs.V4OVNMasqueradeIP.String(), err)
+		}
+		subnets = append(subnets, masqIPNet)
+		neighborIPs = append(neighborIPs, staleMasqueradeIPs.V4OVNMasqueradeIP, staleMasqueradeIPs.V4DummyNextHopMasqueradeIP)
+		if err := nodeipt.DelRules(getStaleMasqueradeIptablesRules(staleMasqueradeIPs.V4OVNMasqueradeIP)); err != nil {
+			return fmt.Errorf("failed to delete forwarding iptables rules for stale masquerade subnet %s: ", err)
+		}
+	}
+
+	if config.IPv6Mode && staleMasqueradeIPs.V6HostMasqueradeIP != nil {
+		// Delete any stale masquerade IP from external bridge.
+		hostMasqIPNet, err := util.LinkAddrGetIPNet(link, staleMasqueradeIPs.V6HostMasqueradeIP)
+		if err != nil {
+			return fmt.Errorf("unable to get IPNet from link %s: %v", link, err)
+		}
+		if hostMasqIPNet != nil {
+			if err := util.LinkAddrDel(link, hostMasqIPNet); err != nil {
+				return fmt.Errorf("failed to remove masquerade IP from bridge %s: %w", link, err)
+			}
+		} else {
+			return nil
+		}
+		_, masqIPNet, err := net.ParseCIDR(fmt.Sprintf("%s/128", staleMasqueradeIPs.V6OVNMasqueradeIP.String()))
+		if err != nil {
+			return fmt.Errorf("failed to parse V6OVNMasqueradeIP %s: %v", staleMasqueradeIPs.V6OVNMasqueradeIP.String(), err)
+		}
+		subnets = append(subnets, masqIPNet)
+		neighborIPs = append(neighborIPs, staleMasqueradeIPs.V6OVNMasqueradeIP, staleMasqueradeIPs.V6DummyNextHopMasqueradeIP)
+		if err := nodeipt.DelRules(getStaleMasqueradeIptablesRules(staleMasqueradeIPs.V6OVNMasqueradeIP)); err != nil {
+			return fmt.Errorf("failed to delete forwarding iptables rules for stale masquerade subnet %s: ", err)
+		}
+	}
+
+	for _, ip := range neighborIPs {
+		if err := util.LinkNeighDel(link, ip); err != nil {
+			klog.Warningf("Failed to remove IP neighbour entry for ip %s, on iface %s: %v",
```

Comment:
Why are we just loging a warning here and return an error for all other del operations?

Reply:
agh, sorry sorry. I should return error here too. Thanks so much Patryk.